paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewState.kt

```diff
@@ -85,6 +85,7 @@ internal sealed class CustomerSheetViewState(
         val primaryButtonEnabled: Boolean,
         val customPrimaryButtonUiState: PrimaryButton.UIState?,
         val mandateText: String? = null,
+        val requiresMandate: Boolean = false,
```

Comment:
I think having 2 fields to represent the same thing will lead to confusion here.

I'd expect mandateText to be null if requiresMandate is false. Why do we need 2 fields for this?

Reply:
I agree with you, having two fields to represent the mandate text is not the most ideal. The mandate text is being emitted by the USBankAccountViewModel, which we handle in `onMandateTextChanged`. Because of this it is a bit more difficult to generate the mandate text on the fly when we need it. In this instance, when we change the payment method forms, the mandate should not show up when on card, but when the user moves back to us bank account, we need to show it again, but the callback does not get triggered since we aren't retrieving any results, instead we are using the persisted form data. Keeping the mandate text not null and deciding when to show it makes things simpler. Hope that makes sense.

Reply:
What happens when we add support for Sepa (or another LPM that requires mandates). Would we also set the text in this field? Would that mean switching from Card -> Sepa -> US Bank would be another bug where us bank doesn't show the mandate?

Reply:
True, I have addressed this in the latest commit. Thanks!